explorer/src/components/account/AnchorAccountCard.tsx

```diff
@@ -0,0 +1,176 @@
+import React from "react";
+
+import { Account } from "providers/accounts";
+import { Address } from "components/common/Address";
+import { BorshAccountsCoder } from "@project-serum/anchor";
+import { capitalizeFirstLetter } from "utils/anchor";
+import { ErrorCard } from "components/common/ErrorCard";
+import { PublicKey } from "@solana/web3.js";
+import BN from "bn.js";
+
+import ReactJson from "react-json-view";
+import { useCluster } from "providers/cluster";
+import { useAnchorProgram } from "providers/anchor";
+
+export function AnchorAccountCard({ account }: { account: Account }) {
+  const { url } = useCluster();
+  const program = useAnchorProgram(
+    account.details?.owner.toString() ?? "",
+    url
+  );
+
+  if (!program) {
+    return (
+      <ErrorCard text={"Could not decode anchor program for this address"} />
+    );
+  }
+
+  if (!account.details || !account.details.rawData) {
+    return (
+      <ErrorCard text={"This account is parsed as an SPL-native account"} />
+    );
+  }
+  const accountBuffer = account.details.rawData;
+
+  const discriminator = accountBuffer.slice(0, 8) ?? undefined;
+  if (!discriminator) {
+    return <ErrorCard text={"Failed to find anchor account discriminator"} />;
+  }
+
+  let foundAccountLayoutName: string | undefined;
+  let decodedAnchorAccountData: Object | undefined;
+  if (program) {
+    // Iterate all the structs, see if any of the name-hashes match
+    Object.keys(program.account).forEach((accountType) => {
+      const layoutName = capitalizeFirstLetter(accountType);
+      const discriminatorToCheck =
+        BorshAccountsCoder.accountDiscriminator(layoutName);
+
+      if (discriminatorToCheck.equals(discriminator)) {
+        foundAccountLayoutName = layoutName;
+        const accountDecoder = program.account[accountType];
+        decodedAnchorAccountData = accountDecoder.coder.accounts.decode(
+          layoutName,
+          accountBuffer
+        );
+      }
+    });
+  }
```

Comment:
Whenever you write something like this, imagine React running it every time you hover your mouse over the surrounding UI. If the calculation of `decodedAnchorAccountData` is expensive, or creates then throws away many intermediate objects/arrays/strings, you might consider `useMemo` to compute it _once_.

Reply:
Thanks! Wrapped with useMemo in latest commit.